modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/serviceextensions/preupdatepassword/PreUpdatePasswordActionFailureTestCase.java

```diff
@@ -328,7 +337,66 @@ public void testUserSetPasswordViaOfflineInviteLink() throws Exception {
+
+        // Ideally this need to come 500 but due to a bug this comes 202
+        // Related Issue: https://github.com/wso2/product-is/issues/25668
+        assertNotNull(response);
+        String status = response.get("status").toString();
+        assertEquals(status, String.valueOf(expectedPasswordUpdateResponse.getStatusCode()));
+        assertEquals(response.get("scimType"), String.valueOf(expectedPasswordUpdateResponse.getErrorMessage()));
+        assertTrue(response.get("detail").toString().contains(expectedPasswordUpdateResponse.getErrorDetail()));
+        assertActionRequestPayload(null, TEST_USER_PASSWORD, PreUpdatePasswordEvent.FlowInitiatorType.ADMIN,
+                PreUpdatePasswordEvent.Action.REGISTER);
+    }
+
+    @Test(dependsOnMethods = "testAdminInitiatedUserRegistration",
+            description = "Verify the application initiated user registration with pre update password action failure")
+    public void testApplicationInitiatedUserRegistration() throws Exception {
+
+        String token = getTokenWithClientCredentialsGrant(application.getId(), clientId, clientSecret);
+        UserObject appRegisteredUserInfo = new UserObject()
+                .userName(TEST_USER2_USERNAME)
+                .password(TEST_USER_PASSWORD)
+                .name(new Name().givenName(TEST_USER_GIVEN_NAME).familyName(TEST_USER_LASTNAME))
+                .addEmail(new Email().value(TEST_USER_EMAIL));
+        org.json.simple.JSONObject response = scim2RestClient.createUser(appRegisteredUserInfo, token);
+
+        // Ideally this need to come 500 but due to a bug this comes 202
+        // Related Issue: https://github.com/wso2/product-is/issues/25668
+        assertNotNull(response);
+        String status = response.get("status").toString();
+        assertEquals(status, String.valueOf(expectedPasswordUpdateResponse.getStatusCode()));
+        assertEquals(response.get("scimType"), String.valueOf(expectedPasswordUpdateResponse.getErrorMessage()));
+        assertTrue(response.get("detail").toString().contains(expectedPasswordUpdateResponse.getErrorDetail()));
+        assertActionRequestPayload(null, TEST_USER_PASSWORD, PreUpdatePasswordEvent.FlowInitiatorType.APPLICATION,
+                PreUpdatePasswordEvent.Action.REGISTER);
+    }
+
+    @Test(dependsOnMethods = "testApplicationInitiatedUserRegistration",
+            description = "Verify the user initiated registration with pre update password action failure")
+    public void testUserInitiatedUserRegistration() throws Exception {
+
+        Object responseObj = flowExecutionClient.initiateFlowExecution(REGISTRATION_FLOW_TYPE);
+
+        FlowExecutionRequest flowExecutionRequest = buildUserRegistrationFlowRequest();
+        Object executeResponseObj = flowExecutionClient.executeFlow(flowExecutionRequest);
+        assertTrue(executeResponseObj instanceof Error, "Unexpected response type for flow execution.");
+
```

Comment:
Dont we need to validate the error code, error message and error description in the Error object ?